opentasks-provider/src/main/java/org/dmfs/provider/tasks/TaskDatabaseHelper.java

```diff
@@ -783,6 +786,12 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion)
             db.execSQL(createIndexString(Tables.INSTANCES, false, TaskContract.Instances.INSTANCE_START_SORTING));
             db.execSQL(createIndexString(Tables.INSTANCES, false, TaskContract.Instances.INSTANCE_DUE_SORTING));
         }
+        if (oldVersion < 17)
+        {
+            // Note: in general only a small fraction of all rows will have this flag set, so an index can improve performance on large tables
+            db.execSQL("alter table " + Tables.TASKS + " add column " + Tasks.INSTANCES_STALE + " INTEGER DEFAULT 1;");
+            db.execSQL(createIndexString(Tables.TASKS, false, Tasks.INSTANCES_STALE));
```

Comment:
Maybe run `UpdateInstancesTask` right away here to update all existing instances?